plugins/extra/karma.go

```diff
@@ -60,25 +118,22 @@ func (p *karmaPlugin) cleanedName(name string) string {
 
 // GetKarmaFor returns the karma for the given name.
 func (p *karmaPlugin) GetKarmaFor(name string) int {
-	out := &KarmaTarget{Name: p.cleanedName(name)}
-
-	_ = p.db.View(func(tx *nut.Tx) error {
-		bucket := tx.Bucket("karma")
-		return bucket.Get(out.Name, out)
-	})
-
+	out := &Karma{Name: p.cleanedName(name)}
+	_, _ = p.db.Get(out)
 	return out.Score
 }
 
 // UpdateKarma will update the karma for a given name and return the new karma value.
 func (p *karmaPlugin) UpdateKarma(name string, diff int) int {
-	out := &KarmaTarget{Name: p.cleanedName(name)}
+	out := &Karma{Name: p.cleanedName(name)}
```

Comment:
Dumb question: does this default the ID to be autoincrement and the karma value to be 0?

Reply:
Yes. This uses go to set them to the zero value (which for ints is 0). In the transaction we try to find it based on the name and insert a blank value if it's not found then update it with the changed score.